test/test_navigation.js

```diff
@@ -0,0 +1,16 @@
+/* global describe, it */
+let generatePageTree = require("../lib/tree");
+let Navigation = require("../lib/navigation");
+let { assertSame, fixturesPath, fixturesDir } = require("./util");
+
+describe("navigation", () => {
+	it("should generate a navigation", async () => {
+		let descriptors = require(fixturesPath("pages.js"));
+		let tree = generatePageTree(descriptors, fixturesDir);
+		await Promise.all(tree.map(async page => page.load()));
```

Comment:
this seems smelly; feels like that should be encapsulated elsewhere (e.g. within a `Tree#children` getter; `await tree.children`)

Reply:
We can refactor that later.